src/types.rs

```diff
@@ -240,6 +240,60 @@ impl BlockTime {
     }
 }
 
+/// Balance differentiated in various categories
+#[derive(Serialize, Deserialize, Debug, PartialEq, Clone, Default)]
+pub struct Balance {
+    /// All coinbase outputs not yet matured
+    pub immature: u64,
+    /// Unconfirmed UTXO generated by a wallet tx
+    pub trusted_pending: u64,
+    /// Unconfirmed UTXO received from an external wallet
+    pub untrusted_pending: u64,
+    /// All rest available balance
+    pub available: u64,
+}
+
+impl Balance {
```

Comment:
I would also add a `get_total()` here because just getting the whole balance is probably something most people will want to do